project-base/storefront/components/Pages/Order/TransportAndPayment/TransportAndPaymentSelect/TransportAndPaymentSelect.tsx

```diff
@@ -309,3 +201,13 @@ const ResetButton: FC<ResetButtonProps> = ({ text, dataTestId, onClick }) => (
         <ArrowIcon className="ml-2" />
     </button>
 );
+
+const getPickupPlaceDetail = (
+    selectedTransport: Maybe<TransportWithAvailablePaymentsAndStoresFragmentApi>,
+    selectedPickupPlace: ListedStoreFragmentApi | null,
+    transportItem: TransportWithAvailablePaymentsAndStoresFragmentApi,
+) =>
+    selectedTransport?.uuid === transportItem.uuid &&
+    transportItem.stores?.edges?.some((storeEdge) => storeEdge?.node?.identifier === selectedPickupPlace?.identifier)
+        ? selectedPickupPlace
+        : null;
```

Comment:
Can we work with `undefined` and `null` use only when it's necessary for the types which usually only comes from BE API?

Reply:
`pickupPlace` itself can be null. To accommodate for only working with undefined, I had to add the exclamation mark:

<img width="852" alt="image" src="https://github.com/shopsys/shopsys/assets/55181354/d875e56d-6c87-47e3-a919-d23a4e7c5ce3">